web/packages/core/src/internal/player/inner.tsx

```diff
@@ -2245,6 +2177,80 @@ export function isFallbackElement(elem: Element): boolean {
     return false;
 }
 
+/**
+ * Prompt the user to download a file.
+ *
+ * @param blob The content to download.
+ * @param name The name to give the file.
+ */
+function saveFile(blob: Blob, name: string): void {
+    const blobURL = URL.createObjectURL(blob);
+    const link = document.createElement("a");
+    link.href = blobURL;
+    link.download = name;
+    link.click();
+    URL.revokeObjectURL(blobURL);
+}
+
+/**
+ * Create a new Uint8Array object from a base64-encoded string.
+ *
+ * @param bytesBase64 The base64-encoded string.
+ * @returns The new Uint8Array.
+ */
+function base64ToArray(bytesBase64: string): Uint8Array {
+    const byteString = atob(bytesBase64);
+    return Uint8Array.from(byteString, char => char.charCodeAt(0));
+}
+
+/**
+ * Create a new Blob of the given type from a base64-encoded string.
+ *
+ * @param bytesBase64 The base64-encoded string..
+ * @returns The new Blob.
+ */
+function base64ToBlob(bytesBase64: string, mimeString: string): Blob {
+    const ab = base64ToArray(bytesBase64);
+    const blob = new Blob([ab], { type: mimeString });
+    return blob;
+}
+
+/**
+ * Check if string is a base-64 encoded SOL file.
+ *
+ * @param solData The base-64 encoded SOL string.
+ * @returns If the string represent a base-64 encoded SOL file.
+ */
+function isB64SOL(solData: string): boolean {
+    try {
+        const decodedData = atob(solData);
+        return isSolData(decodedData);
+    } catch (e) {
+        return false;
+    }
+}
+
+/**
+ * Check if string is structured like SOL data up until the end of the header.
+ * See https://www.sans.org/blog/local-shared-objects-aka-flash-cookies/.
+ *
+ * @param data The SOL string.
+ * @returns If the string seemingly represents an SOL file.
+ */
+function isSolData(data: string): boolean {
+    return (
+       // First two bytes are a magic value (0x00 0xbf)
+       data.charCodeAt(0) === 0x00 && data.charCodeAt(1) === 0xbf &&
+       // Next four bytes give the length of the LSO file excluding the first six bytes.
+       // First byte code * 256^3 + second byte code * 256^2 + third byte code * 256^1 + fourth byte code * 256^0 (1) + 6
+       [0, 1, 2, 3].reduce((acc, i) => acc + data.charCodeAt(i + 2) * (256 ** (3 - i)), 6) === data.length &&
+       // Next four bytes are another magic value (ASCII value of TCSO)
+       data.slice(6, 10) === "TCSO" &&
+       // Next six bytes are padding (0x00 0x04 0x00 0x00 0x00 0x00)
+       [0x00, 0x04, 0x00, 0x00, 0x00, 0x00].every((v, i) => data.charCodeAt(10 + i) === v)
+    );
+}
```

Comment:
I think just checking the first two bytes should be enough. to be honest.
I'm a little worried that bugs in our encoder would make a save file not appear (perhaps we write more than we say we do), and in such a situation I'd rather see (and debug) an invalid save file than just have it silently not show up (even though ruffle will still try to use it)

Reply:
I think I'd rather check the first two and TSC0, just the first two seems like too little since this loops through the entire local storage so it doesn't seem that unlikely for there to be false positives on sites that use local storage extensively if just checking the first two.

Reply:
That sounds good. I like the hardcoded magic values, but anything calculated feels like it could cause issues one day